### PR TITLE
fix(tracing): collect logs only if call succeeded in geth `callTracer`

### PR DESCRIPTION
**Pull request by ArtificialPB**
Logs should only be collected for calls where it and all of its parents succeeded. 

This logic is ported from geth `callTracer`: https://github.com/ethereum/go-ethereum/blob/master/eth/tracers/native/call.go#L278-L289

### CODE REVIEW

File: src/tracing/builder/geth.rs [hunk 1]
```diff
@@ -169,6 +172,18 @@ impl GethTraceBuilder {
...
+    /// Returns true if the given trace or any of its parents failed.
+    fn call_or_parent_failed(&self, node: &CallTraceNode) -> bool {
+        if node.trace.is_error() {
+            return true;
+        }
+
+        if let Some(parent) = node.parent {
+            return self.call_or_parent_failed(&self.nodes[parent]);
+        }
+        false
```
Comment (mattsse) [inline src/tracing/builder/geth.rs:184]: can we solve this iteratively by looping instead?

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, 

I'd like to use `while let Some` here

File: src/tracing/builder/geth.rs (hunk 1 above)
Comment (ArtificialPB) [reply src/tracing/builder/geth.rs]: done

**Review by mattsse — APPROVED** [review]
lgtm, ty!

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**